src/webserver/DOSGuard.h

```diff
@@ -54,11 +57,44 @@ class DOSGuard
             return true;
 
         std::unique_lock lck(mtx_);
-        auto it = ipFetchCount_.find(ip);
-        if (it == ipFetchCount_.end())
-            return true;
+        bool fetchesOk = true;
+        bool connsOk = true;
+        {
+            auto it = ipFetchCount_.find(ip);
```

Comment:
Consider immediately invoked lambda or a helper function for this. 
If you only have one map it could return a pair:
```
auto [fetchesOk, connsOk] = [this] { 
    auto it = talley_.find(ip); 
    return ...; 
}();
```